jscomp/others/js_json.ml

```diff
@@ -60,7 +61,28 @@ let test (type a) (x : 'a) (v : a kind) : bool =
   | Array -> Js_array.isArray x 
   | Object -> (Obj.magic x) != Js.null && Js.typeof x = "object" && not (Js_array.isArray x )
 
-
 external parse : string -> t = "JSON.parse" [@@bs.val]
 external stringifyAny : 'a -> string option = "JSON.stringify" [@@bs.val] [@@bs.return undefined_to_opt]
 (* TODO: more docs when parse error happens or stringify non-stringfy value *)
+
+external null : t = "" [@@bs.val]
+
+external string : string -> t = "%identity"
+
+external number : float -> t = "%identity"
+
+external boolean : Js.boolean -> t = "%identity" 
+
+external object_ : t Js_dict.t -> t = "%identity"
+
+external array_ : t array -> t = "%identity"
```

Comment:
@bobzhang what's the convention? `_array` or `array_`? We've gone with the former because it corresponds to name mangling for obj